### PR TITLE
Test GitRepo detail view displays incorrect number of bundles + code improvements and fixes

### LINKED ISSUE #9866: GitRepo detail view displays incorrect number of bundles
To repo:
1. Create a non-empty gitrepo named `test`
2. Create another non-empty gitrepo `test-abc`
3. Go to the detail view for the first repo - # of bundles is wrong

Found while investigating SURE-6692

### PR DESCRIPTION
**Pull request by aalves08**
### Summary
Fixes #9866 

### Occurred changes and/or fixed issues

- add cluster tab to git repo details 
- improve code on fleet dashboard view (better code abstraction + adding missing data fetches, etc) 
- fix count of clusters in fleet dashboard view badges (`targetClustersReady` prop)
- add e2e test for 9866

### Areas or cases that should be tested

- Follow repro steps of #9866 and check that the following views in Fleet all work fine: Dashboard view, Git Repo list view, Git Repo detail view, Git Repo edit view.

**Note:** if you are having problems finding proper examples of git repos to add in order to test functionality, just check what was done in `gitrepo.spec.ts` and re-create those manually on Rancher Dashboard. They use a repo called `fleet-examples` and work just fine (this is repo is also mentioned in the official docs for Fleet)

### Areas which could experience regressions

Mentioned in the previous point

> - Follow repro steps of #9866 and check that the following views in Fleet all work fine: Dashboard view, Git Repo list view, Git Repo detail view, Git Repo edit view.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

File: cypress/e2e/tests/pages/fleet/gitrepo.spec.ts [hunk 1]
```diff
@@ -94,33 +96,102 @@ describe('Git Repo', { tags: ['@fleet', '@adminUser'] }, () => {
...
+
+      listPage.waitForPage();
+      listPage.selectWorkspace('fleet-local');
+
+      listPage.goToDetailsPage(basicRepos[1].name);
+
+      const gitRepoDetails = new FleetGitRepoDetailsPo('fleet-local', basicRepos[1].name);
+
+      gitRepoDetails.waitForPage();
+      gitRepoDetails.gitRepoTabs().clickTabWithSelector('#bundles');
```
Comment (izaac) [inline cypress/e2e/tests/pages/fleet/gitrepo.spec.ts:170]: This tab/button has a `data-testid` we could use it.

File: cypress/e2e/tests/pages/fleet/gitrepo.spec.ts [hunk 2]
```diff
@@ -94,33 +96,102 @@ describe('Git Repo', { tags: ['@fleet', '@adminUser'] }, () => {
...
+      gitRepoCreatePage.setGitRepoUrl(basicRepos[1].repo);
+      gitRepoCreatePage.setBranchName(basicRepos[1].branch);
+      gitRepoCreatePage.gitRepoPaths().setValueAtIndex(basicRepos[1].path, 0);
+      gitRepoCreatePage.goToNext();
+      gitRepoCreatePage.create();
+
+      listPage.waitForPage();
+      listPage.selectWorkspace('fleet-local');
+
+      listPage.goToDetailsPage(basicRepos[1].name);
```
Comment (izaac) [inline cypress/e2e/tests/pages/fleet/gitrepo.spec.ts:165]: The issue's steps linked in the PR describes to go navigate back to the first gitRepo. Or is it that any gitrepo will behave the same without the fix.

**Review by izaac — CHANGES_REQUESTED** [review]

File: cypress/e2e/tests/pages/fleet/gitrepo.spec.ts (hunk 2 above)
Comment (mantis-toboggan-md) [reply cypress/e2e/tests/pages/fleet/gitrepo.spec.ts:165]: I can answer this since I filed that issue: the bug was reproducible with any repos whose name shared the same prefix followed by a hyphen. Since they both start with `e2e-`, the `e2e-git-repo1-test-bundle-count` `e2e-git-repo2-test-bundle-count`  detail views would have each showed the bundle count from both (2) instead of the correct count (1). So this should be okay.

**Review by mantis-toboggan-md — CHANGES_REQUESTED** [review]
I had a couple questions about issue tracking but the code refactoring in the PR looks good to me.

File: shell/detail/fleet.cattle.io.gitrepo.vue [hunk 3]
```diff
@@ -125,6 +142,21 @@ export default {
       class="mt-20"
       :need-related="false"
     >
+      <Tab
```
Comment (mantis-toboggan-md) [inline shell/detail/fleet.cattle.io.gitrepo.vue:145]: Is there an issue to track the addition of this tab? It seems outside the scope of #9866. Semi-related, the table formatting looks off to me: 
Screenshot 2024-05-16 at 9 51 45 AM

Should I file a separate issue for that?

Comment (aalves08) [reply shell/detail/fleet.cattle.io.gitrepo.vue:145]: So, all of this comes from this PR: https://github.com/rancher/dashboard/pull/10776 which is tied to https://github.com/rancher/dashboard/issues/10339. I found some problems within Fleet regarding counts, interfaces throwing errors and the code seemed a bit too much so I went ahead and added the improvements that I felt were needed.

You're right, there's something off when looking at that table in terms of formatting. Never payed attention to that... 🤔  good to have another set of 👀  🙏 

I am going to remove this tab from this PR... I quite out of scope. Also, thinking about it, RBAC might also limit the clusters a user can "see" so I don't think it's needed right now. I can create an issue to track it and maybe we add it in the future.

Thanks for the review!

**Review by izaac — APPROVED** [review]
LGTM :shipit:

File: cypress/e2e/tests/pages/fleet/gitrepo.spec.ts (hunk 2 above)
Comment (izaac) [reply cypress/e2e/tests/pages/fleet/gitrepo.spec.ts:165]: Ah makes sense thanks @mantis-toboggan-md

**Comment by aalves08** [comment]
@mantis-toboggan-md if you're ok with the PR, I'll need your thumbs up as well, otherwise I cannot merge this 😛 Cheers!

**Review by mantis-toboggan-md — APPROVED** [review]
Sorry @aalves08 approved now
